lmfdb/genus2_curves/web_g2c.py

```diff
@@ -634,9 +644,16 @@ def by_label(label):
                 g2c_logger.warning("No rational points data for genus 2 curve %s found in database." % label)
         else:
             ratpts = {}
-        return WebG2C(curve, endo, tama, ratpts, is_curve=(len(slabel)==4))
-
-    def make_object(self, curve, endo, tama, ratpts, is_curve):
+        clus = []
+        for x in tama:
+            if x['p'] != 2:
+                clusentry = db.cluster_pictures.lucky({"label": x['cluster_label']})
+                #clusimg = clusentry['image']
+                clusthmb = clusentry['thumbnail']
```

Comment:
If the data is missing, this `KeyError` will likely crash the server.  Is `cluster_label` ever not computed?  Should you check to see if `clusentry` is `None`?

Reply:
It is not supposed to be not computed, but I can't promise that future people will always compute it for new curves. So you seem to suggest that I should throw some other error in case the key does not exist, or in case clusentry is empty. I will see if I can find another place where this is done and copy that approach.

Reply:
Okay, I modified this now to fit with the strategy used in the rest of web_g2c.py. If the code to find the cluster data fails, it will throw a KeyError with an explanation.